crates/gen/src/element_type.rs

```diff
@@ -0,0 +1,165 @@
+use crate::blob::Blob;
+use crate::codes::{Decode, TypeDefOrRef};
+use crate::signatures::TypeSig;
+
+#[allow(dead_code)]
+pub enum ElementType {
+    Void,
+    Boolean,
+    Char,
+    I1,
+    U1,
+    I2,
+    U2,
+    I4,
+    U4,
+    I8,
+    U8,
+    R4,
+    R8,
+    String,
+
+    Ptr(TypeSig),
```

Comment:
Please remove everything that isn't specific to WinRT support. The ECMA-335 spec is enormous and the rabbit hole is long. 😉 

Let's just add what we need when we need it as we cannot possibly test all of these permutations and I don't want to have to support it.

Reply:
Good call. This also reduces the surface area of this change a lot, because none of the signature portion is necessary any longer, so I removed all of that until its necessary as well.